Magento_Theme/templates/custom_footer.phtml

```diff
@@ -0,0 +1,6 @@
+<?php $block = $this->getLayout()->getBlock('custom.footer'); ?>
+
+<div>
+    <h2> <?php echo $block->getData('custom_footer_text'); ?> </h2>
```

Comment:
when you have to display some text that has to be dynamically inserted, like PHP functions, XML arguments, translations, etc. You always have to wrap this to the escaping function. If you text inside a tag element, like 
```html
<tag>text</tag>
```
use the escaping function `escapeHtml`.
```html
<h2><?= $escaper->escapeHtml($block->getData('custom_footer_text')); ?></h2>
```
help IDE to find the class also via `@var`
```php
/**
 * @var Magento\Framework\View\Element\Template $block
 * @var Magento\Framework\Escaper $block
 */
```